### PR TITLE
Dylan

### PR DESCRIPTION
**Pull request by dmcdo**

### CODE REVIEW

File: src/containers/exceptions.py [hunk 1]
```diff
@@ -10,6 +10,7 @@
 from pexpect import ExceptionPexpect
 
 from src.system.syspath import get_server_log_file
+from src.system.socket import ClientServerSocket
```
Comment (Kippiii) [inline src/containers/exceptions.py:13]: I'm surprised this does not cause circular recursion...

File: src/containers/exceptions.py [hunk 2]
```diff
@@ -181,18 +182,16 @@ def __str__(self):
         return f"The path {self.path} does not exist"
 
 
-def get_server_error(value: str, sock: "ClientServerSocket") -> None:
```
Comment (Kippiii) [inline src/containers/exceptions.py:184]: Why did you move this?

**Review by Kippiii — CHANGES_REQUESTED** [review]

File: src/cli/cli.py [hunk 3]
```diff
@@ -103,6 +111,22 @@ def help(self, cmd: List[str]) -> None:  # pylint: disable=unused-argument
...
 
+    def sftp(self, cmd: List[str]) -> None:
+        name = cmd[0]
+        comp = re.compile(CONTAINER_NAME_REGEX)
+        if not comp.match(name):
+            self.out_stream.write(f"'{name}' is not a valid container name\n")
+            return
+        self.container_manager.sftp(name)
+
+    def view_files(self, cmd: List[str]) -> None:
```
Comment (Kippiii) [inline src/cli/cli.py:123]: If I run this on my install without Filezilla installed, it gives unhelpful exception:

```FileNotFoundError: [Errno 2] No such file or directory: PosixPath('/home/fcorp/Documents/Jabberwocky/container_manager/contrib/filezilla/bin/filezilla')```

File: src/cli/cli.py [hunk 4]
```diff
@@ -295,6 +315,14 @@ def server_halt(self, cmd: List[str]) -> None:  # pylint: disable=unused-argumen
         """
         self.container_manager.server_halt()
 
+    def server_panic(self, cmd: List[str]) -> None:  # pylint: disable=unused-argument
```
Comment (Kippiii) [inline src/cli/cli.py:327]: Running panic gives this error:

```File "/home/fcorp/Documents/Jabberwocky/container_manager/src/containers/container_manager_server.py", line 153, in start_connection```

File: download_prerequisites.py [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+from sys import platform
+from os import listdir, rmdir, remove, makedirs
+
+CONTRIB_PATH = Path(__file__).parent / "contrib"
+
+if not CONTRIB_PATH.exists():
+    makedirs(CONTRIB_PATH)
+
+if not (CONTRIB_PATH / "filezilla").exists():
+    print("Downaloding Filezilla...")
```
Comment (Kippiii) [inline download_prerequisites.py]: Misspelling

File: src/containers/exceptions.py (hunk 2 above)
Comment (dmcdo) [reply src/containers/exceptions.py:184]: It fixes the circular import and I think it's a logical place to put the function.
If anything it should probably be a member of ClientServerSocket but it doesn't matter enough to change.

File: src/cli/cli.py (hunk 4 above)
Comment (dmcdo) [reply src/cli/cli.py:327]: Full error message?

Comment (Kippiii) [reply src/cli/cli.py:327]: I can't replicate this issue anymore :(

**Review by Kippiii — APPROVED** [review]

**Pull request merged**